### PR TITLE
RavenDB-7592: better looking cli

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Raven.Server/Utils/RavenCli.cs [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+
+            return true;
+        }
+
+        private static bool CommandLog(List<string> args)
+        {
+            switch (args.First())
+            {
+                case "on":
+                    LoggingSource.Instance.EnableConsoleLogging();
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:208]: This won't work with the Pipe

File: src/Raven.Server/Utils/RavenCli.cs [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+            Console.ResetColor();
+            Console.WriteLine();
+            Console.Write("Are you sure you want to quit the server ? [y/N] : ");
+            Console.Out.Flush();
+
+            var k = Console.ReadKey();
+            Console.Out.Flush();
+
+            Console.WriteLine();
+            if (k.KeyChar.Equals('Y') ||
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:85]: Simplify this

File: src/Raven.Server/Utils/RavenCli.cs [hunk 3]
```diff
@@ -0,0 +1,706 @@
...
+                        }
+                        break;
+
+                    default:
+                        msg.Append(l);
+                        break;
+                }
+            }
+            if (test == false)
+                Console.Write(msg);
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:63]: won't work on Pipe

File: src/Raven.Server/Utils/RavenCli.cs [hunk 4]
```diff
@@ -0,0 +1,706 @@
...
+
+        private class SingleAction
+        {
+            public int NumOfArgs;
+            public Func<List<string>, bool> DelegateFync;
+        }
+
+        private static bool CommandQuit(List<string> args)
+        {
+            Console.ResetColor();
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:76]: In general, all the console stuff should use a TextWriter instead.

File: src/Raven.Server/Utils/RavenCli.cs [hunk 5]
```diff
@@ -0,0 +1,706 @@
...
+                case 2:
+                    GC.Collect(GC.MaxGeneration);
+                    break;
+                default:
+                    WriteError("Invalid argument passed to GC. Can be 0, 1 or 2");
+                    return false;
+            }
+
+            GC.WaitForPendingFinalizers();
+            Console.ForegroundColor = ConsoleColor.Green;
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:191]: Include time taken here

File: src/Raven.Server/Utils/RavenCli.cs [hunk 6]
```diff
@@ -0,0 +1,706 @@
...
+        private static bool CommandInfo(List<string> args)
+        {
+            var memoryInfo = MemoryInformation.GetMemoryInfo();
+            Console.ForegroundColor = ConsoleColor.Cyan;
+            Console.WriteLine(" Build {0}, Version {1}, SemVer {2}, Commit {3}\r\n PID {4}, {5} bits, {6} Cores, Arch: {9}\r\n {7} Physical Memory, {8} Available Memory",
+                ServerVersion.Build, ServerVersion.Version, ServerVersion.FullVersion, ServerVersion.CommitHash, Process.GetCurrentProcess().Id,
+                IntPtr.Size * 8, ProcessorInfo.ProcessorCount, memoryInfo.TotalPhysicalMemory, memoryInfo.AvailableMemory, RuntimeInformation.OSArchitecture);
+
+            var bitsNum = IntPtr.Size * 8;
+            if (bitsNum == 64 && _server.Configuration.Storage.ForceUsing32BitsPager)
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:239]: Why not always put the mode?

File: src/Raven.Server/Utils/RavenCli.cs [hunk 7]
```diff
@@ -0,0 +1,706 @@
...
+            if (args == null || args.First().Equals("no-clear") == false)
+                Console.Clear();
+            WelcomeMessage.Print();
+            return true;
+        }
+
+        private static bool CommandLowMem(List<string> args)
+        {
+            Console.ResetColor();
+            Console.Write("Sendin Low Memory simulation signal... ");
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:260]: Typo

File: src/Raven.Server/Utils/RavenCli.cs [hunk 8]
```diff
@@ -0,0 +1,706 @@
...
+            Console.Out.Flush();
+            LowMemoryNotification.Instance.SimulateLowMemoryNotification();
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine("Sent.");
+            Console.ResetColor();
+            Console.Out.Flush();
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:266]: Memory before & after?

File: src/Raven.Server/Utils/RavenCli.cs [hunk 9]
```diff
@@ -0,0 +1,706 @@
...
+                PrintCliHeader();
+                var line = Console.ReadLine();
+                Console.Out.Flush();
+
+                if (line == null)
+                {
+                    Thread.Sleep(75); //waiting for Ctrl+C 
+                    if (ctrlCPressed)
+                        break;
+                    Console.WriteLine("End of standard input detected, switching to server mode...");
```
Comment (ayende) [inline src/Raven.Server/Utils/RavenCli.cs:446]: This happens in Unix when running as daemon, right?

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by ayende** [comment]
Also note that this needs to be checked against our scripts, which check our output

File: src/Raven.Server/Utils/RavenCli.cs (hunk 4 above)
Comment (aviviadi) [reply src/Raven.Server/Utils/RavenCli.cs:76]: ok

File: src/Raven.Server/Utils/RavenCli.cs (hunk 9 above)
Comment (aviviadi) [reply src/Raven.Server/Utils/RavenCli.cs:446]: on unix daemon, in docker, and also in windows nano server or any other console not supporting ReadKey

Comment (ayende) [reply src/Raven.Server/Utils/RavenCli.cs:446]: I think we need to avoid using ReadKey in general

Comment (ayende) [reply src/Raven.Server/Utils/RavenCli.cs:446]: Given that we are going to be just using TextWriter / TextReader

File: src/Raven.Server/Utils/RavenCli.cs (hunk 6 above)
Comment (aviviadi) [reply src/Raven.Server/Utils/RavenCli.cs:239]: we write x64 or x86 always. This is a unique place where running on 64bits enforcing 32bits. And it is a rare one. I am not sure I would want to see "Enforce 32 bit: No" all the time

Comment (ayende) [reply src/Raven.Server/Utils/RavenCli.cs:239]: Oh, I see, okay

**Pull request merged**